### PR TITLE
Improve error handling for target_ids

### PR DESCRIPTION
**Pull request by iainelder**
I passed the account ID of an ex-account and had botocove blow up with a KeyError inside the _generate_account_sessions function.

It's good that it fails, but the user feedback could be improved.

This is my proposal for a new set of tests that checks the behavior of the target_ids.

I've just covered the case I encountered today, but there are surely more that would be good to cover.

### CODE REVIEW

**Comment by connelldave** [comment]
Seems reasonable to me! I can't offhand think of a cleaner way to check without messing around too much - might be just worth adding a comment for clarity is all I can suggest

**Comment by iainelder** [comment]
Today I unintentionally passed an empty list and was confused when botocove iterated over more accounts that I had expected to be in the list.

I've added a new test case for botocove to raise an error when the target IDs list is empty.

 The `target_ids` documentation doesn't say what to expect when the input list is empty, so I think we are allowed to change the behavior.

**Comment by iainelder** [comment]
Unless there are any more cases you think we need to cover here, this for me is ready for review as a first pass of testing this parameter's behavior.

**Comment by iainelder** [comment]
After referring back to test_regions.py, now I think I think I've covered all the use cases for target_ids.

**Comment by connelldave** [comment]
> Today I unintentionally passed an empty list and was confused when botocove iterated over more accounts that I had expected to be in the list.

Just to be certain - the behaviour was the same as if the value was None and org lookup was used?

> The `target_ids` documentation doesn't say what to expect when the input list is empty, so I think we are allowed to change the behavior.

I think this is a regression if my assumption is true - so that also makes fixing it easier. We probably don't have enough users in the world to actually worry about [Hyrum's law](https://www.hyrumslaw.com/) but there's still a chance fixing this breaks someone - I'll write up a note in the changelog for a minor version bump when we merge.

File: botocove/cove_host_account.py [hunk 1]
```diff
@@ -101,6 +101,16 @@ def _generate_account_sessions(self) -> Iterable[CoveSessionInformation]:
         for region in self.target_regions:
             for account_id in self.target_accounts:
                 if self.account_data is not None:
+
+                    # I think this can be true only when running in an organization
+                    # and when target_ids contains an account ID from outside
+                    # the organization. We'd like to validate this earlier, but
+                    # it's not practical without a lot of code reorganization.
```
Comment (connelldave) [inline botocove/cove_host_account.py:108]:
```suggestion

                    # If running with target accounts, but with organization data available
                    # ie running from org master but not targeting whole org
```

**Comment by iainelder** [comment]
> the behaviour was the same as if the value was None and org lookup was used?

Yes, that's what happened. I had intended to pass a list of around 50 accounts, and because of a mistake I actually passed an empty list. So botocove ended up querying all 100-odd accounts in the org.

Thanks for the read about Hyrum's law!

I accepted your shorter comment on the error handling.

I also pushed an extra test to cover an empty regions list.

How does it look now?

**Review by connelldave — APPROVED** [review]
looks good to me! Will release as a minor version.
